src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1890,7 +1890,7 @@ internal override void ReportUnusedImports(SyntaxTree filterTree, DiagnosticBag
                     cancellationToken.ThrowIfCancellationRequested();
 
                     SyntaxTree infoTree = info.Tree;
-                    if (filterTree == null || filterTree == infoTree)
+                    if ((filterTree == null || filterTree == infoTree) && infoTree.Options.DocumentationMode != DocumentationMode.None)
```

Comment:
just to check, this will not affect the IDE right?  i.e. the user cannot select DocMode.None in the IDE?

otherwise, if they do, this change would mean they get *no* unused usings warnings.

Reply:
Correct, the IDE has _never_ passed .None. It has always passed Parse or Diagnose depending on what the underlying flag is, precisely for this reason if I remember correctly.

https://github.com/dotnet/roslyn/blob/62bb55cf33f8b4ce1180e756d0a4864d89d347a0/src/VisualStudio/CSharp/Test/ProjectSystemShim/LegacyProject/CSharpCompilerOptionsTests.cs#L17-L51

Reply:
Interesting.  how does this fix the bug hten?  We'll continue passing along the variable.  But the change i see above in neal's work still won't do the analysis.  So we'll still think the usings are unused... right?

Reply:
Correct: if the user selects DocMode.None, they will get no unused usings reported.

---
In reply to: [184510536](https://github.com/dotnet/roslyn/pull/26425#discussion_r184510536) [](ancestors = 184510536)

Reply:
We now do the analysis and drop the diagnostics on the floor.

---
In reply to: [184529384](https://github.com/dotnet/roslyn/pull/26425#discussion_r184529384) [](ancestors = 184529384)